src/main/java/org/opensearch/knn/index/memory/NativeMemoryCacheManager.java

```diff
@@ -291,8 +291,8 @@ public CacheStats getCacheStats() {
     public NativeMemoryAllocation get(NativeMemoryEntryContext<?> nativeMemoryEntryContext, boolean isAbleToTriggerEviction)
         throws ExecutionException {
         if (!isAbleToTriggerEviction
-            && !cache.asMap().containsKey(nativeMemoryEntryContext.getKey())
-            && maxWeight - getCacheSizeInKilobytes() - nativeMemoryEntryContext.calculateSizeInKB() <= 0) {
+            && (maxWeight - getCacheSizeInKilobytes() - nativeMemoryEntryContext.calculateSizeInKB()) <= 0
+            && !cache.asMap().containsKey(nativeMemoryEntryContext.getKey())) {
```

Comment:
nit: why switch?

Reply:
Since numerical operations are far cheaper than `cache.asMap()`, I intentionally put it ahead of it.